electrode_models/nucleation_single_particle.py

```diff
@@ -0,0 +1,291 @@
+        self.elyte_obj.electric_potential = phi_elyte
+
+        sdot_electron = self.surf_obj.get_net_production_rates(self.bulk_obj)
+        
+        # Faradaic current density is positive when electrons are consumed 
+        # (Li transferred to the electrode)
+        i_Far = -(ct.faraday * sdot_electron)
+        
+        # Calculate the electrolyte species fluxes and associated ionic current 
+        # at the boundary with the separator:
+        N_k_sep, i_io = sep.electrode_boundary_flux(SV, self, params['T']) #?
+
+        if self.name=='anode':
+            # The electric potential of the anode = 0 V.
+            resid[[SVptr['phi_ed'][0]]] = SV_loc[SVptr['phi_ed'][0]]
+            
+        elif self.name=='cathode':
+            # For the cathode, the potential of the cathode must be such that 
+            # the electrolyte electric potential (calculated as phi_ca + 
+            # dphi_dl) produces the correct ionic current between the separator # and cathode:
+            if params['boundary'] == 'current':
+                resid[SVptr['phi_ed']] = i_io - params['i_ext']
+            elif params['boundary'] == 'potential':                  
+                resid[SVptr['phi_ed']] = (SV_loc[SVptr['phi_ed']] 
+                    - params['potential']) 
+
+        # Double layer current has the same sign as i_Far, and is based on 
+        # charge balance in the electrolyte phase:
+        i_dl = self.i_ext_flag*i_io/self.A_surf_ratio - i_Far
+
+        # Differential equation for the double layer potential:
+        resid[SVptr['phi_dl']] = \
+            SVdot_loc[SVptr['phi_dl']] - i_dl*self.C_dl_Inv
+
+        # species production in electrode active material:
+        #sdot_k_ed = self.surf_obj.get_net_production_rates(self.bulk_obj)
+
+        #resid[SVptr['C_k_ed']] = (SVdot_loc[SVptr['C_k_ed']] 
+        #    - self.A_surf_ratio *  sdot_k_ed * self.dyInv / self.eps_solid)
+
+        # Molar production rate of electrode species (kmol/m2/s).
+        sdot_elyte = self.surf_obj.get_net_production_rates(self.elyte_obj)
+        
+        # Double layer current removes Li from the electrolyte.  Subtract this 
+        # from sdot_electrolyte:
+        #sdot_elyte[self.index_Li] -= i_dl / ct.faraday # ? I don't think this is necessary, because the reaction is in the get net production rates
+            
+        # Change in electrolyte species concentration per unit time:
+        dCk_elyte_dt = \
+            ((sdot_elyte * self.A_surf_ratio + self.i_ext_flag * N_k_sep)
+            * self.dyInv / self.eps_elyte) # first term is reaction second term is seperater?
+        resid[SVptr['C_k_elyte']] = SVdot_loc[SVptr['C_k_elyte']] - dCk_elyte_dt
+        #molar production rate of 
+        sdot_cath = self.surf_obj.get_net_production_rates(self.product_obj)
+        # available interface area on carbon particle
+        A_int_avail = self.A_init - self.eps_oxide/self.th_oxide
```

Comment:
So:
1. This line should use the current `eps_oxide` value (`SV_loc[SVptr['eps_oxide']]`), not the initial value.
2. This available area should also be used to scale the other `sdot` values calculated above.  i.e. as `eps_oxide` changes, the `A_surf_ratio` and the area available for `i_Far` will both change.